### DISCUSSION TITLE
Configuring Synchronous Connection Closure in ActiveMQ Artemis analogous to the transport.closeAsync parameter in ActiveMQ Classic

### DISCUSSION THREAD

**Discussion by RomulFobos2** [arkmq-org/arkmq-org-broker-operator#789] [Q&A] [answered]
0

I am in the process of migrating a service from the ActiveMQ Classic message broker to ActiveMQ Artemis and have encountered a specific challenge related to connection management. In ActiveMQ Classic I utilized the transport.closeAsync=false parameter in the transportConnector configuration to ensure synchronous socket closure between the client and the broker. This parameter was critical for ensuring resilience and consistency in message processing within our system.

Transitioning to ActiveMQ Artemis I noticed that there doesn't seem to be a direct analogue to the transport.closeAsync parameter. It appears that ActiveMQ Artemis closes connections asynchronously by default, which could potentially impact the behavior of our application.

Is there a way in ActiveMQ Artemis to configure or adjust the connection closure mechanism to achieve behavior similar to transport.closeAsync=false in ActiveMQ Classic? Can anyone suggest methods or practices for ensuring synchronous connection closure in ActiveMQ Artemis that could help maintain the sequence and reliability of message processing? If anyone has faced a similar challenge or can share their experience with implementing such configurations in ActiveMQ Artemis, I would be grateful for any information or recommendations.

**Comment by brusdev** [comment, marked as answer]
@RomulFobos2 this a question for Apache ActiveMQ Artemis, you can send an email to users@activemq.apache.org
